src/workerd/io/io-context.c++

```diff
@@ -1377,6 +1381,73 @@ kj::Promise<void> IoContext::startDeleteQueueSignalTask(IoContext* context) {
   }
 }
 
+void IoContext::pumpMessageLoop() {
+  kj::Promise<Worker::AsyncLock> asyncLockPromise = nullptr;
+  KJ_IF_SOME(a, actor) {
+    asyncLockPromise = worker->takeAsyncLockWhenActorCacheReady(now(), a, getMetrics());
+  } else {
+    asyncLockPromise = worker->takeAsyncLock(getMetrics());
+  }
+
+  addWaitUntil(asyncLockPromise.then([](Worker::AsyncLock lock) {
+    return lock;
+  }).then([this](Worker::AsyncLock lock) mutable {
+    KJ_REQUIRE(threadId == getThreadId(), "IoContext cannot switch threads");
+    IoContext* previousRequest = threadLocalRequest;
+    KJ_DEFER(threadLocalRequest = previousRequest);
+    threadLocalRequest = nullptr;
+
+    worker->runInLockScope(lock, [&](Worker::Lock& workerLock) {
+      workerLock.requireNoPermanentException();
+
+      auto limiterScope = limitEnforcer->enterJs(workerLock, *this);
+
+      bool gotTermination = false;
+
+      KJ_DEFER({
+        jsg::Lock& js = workerLock;
+        if (gotTermination) {
+          js.terminateExecution();
+        }
+      });
+
+      v8::TryCatch tryCatch(workerLock.getIsolate());
+      try {
+        jsg::Lock& js = workerLock;
+        js.pumpMessageLoop();
+      } catch (const jsg::JsExceptionThrown&) {
```

Comment:
I don't think there's any way for `jsg::JsExceptionThrown` to be thrown by `pumpMessageLoop()`, so can this whole catch block be deleted?

(And also the KJ_DEFER, etc.)

Reply:
Yup will remove it (https://github.com/cloudflare/workerd/pull/3560#discussion_r1971921585)